File: sandGlass/solidSandglass/solidSandglass.py
numRows = 7
numSpaces = 0;   # floor division to get only integer value.
numPatternSymbols = numRows;

patternSymbol = '*'

for rowVal in range(1, numRows + 1):
    for colVal in range(1, numSpaces + 1):
        print("  ", end="")
    for colVal in range(1, numPatternSymbols + 1):
        print(patternSymbol, "", end="") # heer, I am only using this "" because, sep parameter is already giving us one space as sep=" "
    if rowVal <= numRows // 2:
        numPatternSymbols -= 2
        numSpaces += 1
    else:
        numPatternSymbols += 2
        numSpaces -= 1
    print()   # for new line

File: sandGlass/solidSandglass/solidSandglass.cpp
#include <iostream>
using namespace std;

int main() {
    int numRows = 7;
    int rowVal; 
    int colVal; 
    int numSpaces = 0;
    int numPatternSymbols = numRows;

    char patternSymbol = '*';

    for(rowVal = 1; rowVal <= numRows; rowVal++) {
        for(colVal = 1 ;colVal <= numSpaces; colVal++) {
            cout<<"  ";
        } 
        for(colVal = 1; colVal <= numPatternSymbols; colVal++) {
            cout<<patternSymbol<<" ";
        }
        if(rowVal <= numRows / 2) {
            numSpaces++;
            numPatternSymbols -= 2;
        } else {
            numSpaces--;
            numPatternSymbols += 2;
        }
        cout<<"\n";
    }
    return 0;
}
